### ISSUE TITLE
What Data Generation Library Should/Could We Use?

### ISSUE THREAD

**Issue by Adron** [ThrashingCode/colligere#3]
Currently writing docs to describe features/stories, but part of the gist of the CLI app is that it generates data to send against an API end point. Thus, we'll need something that generates good data to send to the API, so which library should we use? Any thoughts, ideas, or any libraries that you've used that we ought to go with?

ping> @olsonso @jinglish @dbriesz

**Comment by jinglish** [comment]
Maybe this one? Seems relatively full-featured--others I've found through some brief Googling are focused entirely on things like names and addresses: https://github.com/icrowley/fake

**Comment by Adron** [comment]
Yeah, the fake by crowley might work. However, there are a few others too that seem to offer different types of data generation. Maybe we should try out a few and play around with them?

**Comment by jinglish** [comment]
I can take that task on, assuming I have some free time this weekend.

**Comment by Adron** [comment]
Cool, feel free to knock it out @jinglish and throw a PR at it!

**Comment by olsonso** [comment]
https://github.com/malisit/kolpa is another that I found. Only can generate for specific instances currently but could be useful depending on what kind of data we are needing to generate

**Comment by Adron** [comment]
Cool, please check it out further @olsonso, we'll need to take a good look at a number of them. I've done a little more searching last night and plan to today. I've gotta get back into getting some of this pre-coding research done!  :)

**Comment by Adron** [comment]
Alright @olsonso @jinglish @dbriesz, I've created a branch called [testing-data-generators](https://github.com/ThrashingCode/colligere/tree/testing-data-generators) so I can start testing out some of these libraries. If y'all have already tested anything lemme know what did didn't work for ya. I'll report back what I get as soon as I wrap up a few tests and all. Might even - if time allows - will blog it too.

**Comment by Adron** [comment]
Ok. So I managed to actually blog my discoveries. Post [Go Library Generation Timings](https://compositecode.blog/2018/06/30/go-library-data-generation-timings/) for the results I came up with. Tried em out synchronously and with some go routines to run them in parallel. Not sure why but Kolpa is really slow when generating lots of data. The fake and faker libs both perfect reasonably well, the later considering it gens a whole struct with JSON tags (properties?, I forget exactly how it's referred to in Go parlance) it does pretty well too. Gonna close this one now and will add the fake lib for now and move forward with some of the next things to implement. Will aim to add some more issues too for some additional low hanging fruit to grab and PR. Cheers
